app/containers/MessageActions.js

```diff
@@ -264,79 +276,72 @@ export default class MessageActions extends React.Component {
 	handleStar = () => {
 		const { actionMessage, toggleStarRequest } = this.props;
 		toggleStarRequest(actionMessage);
-	}
+	};
 
 	handlePermalink = async() => {
 		const { actionMessage } = this.props;
 		const permalink = await this.getPermalink(actionMessage);
 		Clipboard.setString(permalink);
 		showToast(I18n.t('Permalink_copied_to_clipboard'));
-	}
+	};
 
 	handlePin = () => {
 		const { actionMessage, togglePinRequest } = this.props;
 		togglePinRequest(actionMessage);
-	}
+	};
 
 	handleReply = () => {
 		const { actionMessage, replyInit } = this.props;
 		replyInit(actionMessage, true);
-	}
+	};
 
 	handleQuote = () => {
 		const { actionMessage, replyInit } = this.props;
 		replyInit(actionMessage, false);
-	}
+	};
 
 	handleReaction = () => {
 		const { actionMessage, toggleReactionPicker } = this.props;
 		toggleReactionPicker(actionMessage);
-	}
+	};
 
-	handleActionPress = (actionIndex) => {
-		if (actionIndex) {
-			switch (actionIndex) {
-				case this.REPLY_INDEX:
-					this.handleReply();
-					break;
-				case this.EDIT_INDEX:
-					this.handleEdit();
-					break;
-				case this.PERMALINK_INDEX:
-					this.handlePermalink();
-					break;
-				case this.COPY_INDEX:
-					this.handleCopy();
-					break;
-				case this.SHARE_INDEX:
-					this.handleShare();
-					break;
-				case this.QUOTE_INDEX:
-					this.handleQuote();
-					break;
-				case this.STAR_INDEX:
-					this.handleStar();
-					break;
-				case this.PIN_INDEX:
-					this.handlePin();
-					break;
-				case this.REACTION_INDEX:
-					this.handleReaction();
-					break;
-				case this.DELETE_INDEX:
-					this.handleDelete();
-					break;
-				default:
-					break;
-			}
-		}
-		const { actionsHide } = this.props;
-		actionsHide();
+	hideActionSheet() {
+		this.bottomSheetRef.current.snapTo(hiddenBottomSheet);
 	}
 
+	renderHeader = () => (
+		<View style={styles.header}>
+			<View style={styles.panelHeader}>
+				<View style={styles.panelHandle} />
+			</View>
+		</View>
+	);
+
+	renderInner = () => (
+		<View style={[styles.panel, { height: this.bottomSheetHeight }]}>
+			{this.options.map(option => (
+				<Touchable onPress={() => { this.hideActionSheet(); option.handler(); }} key={option.label}>
+					<View style={styles.panelButton}>
+						<CustomIcon name={option.icon} size={18} style={styles.panelButtonIcon} />
+						<Text style={sharedStyles.textRegular}>{option.label}</Text>
+					</View>
+				</Touchable>
+			))}
+		</View>
+	);
+
 	render() {
 		return (
-			null
+			<BottomSheet
```

Comment:
A backdrop is missing.
You can use something like this: https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/views/RoomsListView/SortDropdown.js#L121

Reply:
unfortunately the bottomsheet doesn't have function `onPositionChange` or something similar, but I did manage to do backdrop using `Animated.Code`